packages/core/src/theme/theme.ts

```diff
@@ -231,6 +231,34 @@ export const typography = {
   },
 }
 
+export type ZIndexStrings =
+  | 'auto'
+  | 'absolute'
+  | 'dropdown'
+  | 'sticky'
+  | 'fixed'
+  | 'overlay'
+  | 'onOverlay'
+  | 'offCanvas'
+  | 'modal'
+  | 'popover'
+  | 'tooltip'
```

Comment:
I just made a small [codesandbox](https://codesandbox.io/s/css-types-41cykl?file=/src/App.tsx) to show how we could use the `csstype` library to handle things like this! Let me know what ya think

Reply:
I don't know if that removes the duplication from here. I think Craig is suggesting something like `Object.keys(zIndicies)` but for a typescript type

Reply:
I see! In that case, there are 2 patterns that I'm a fan of. For an object, there's the `keyof typeof {someObject}`, kinda like this:

```typescript
const obj = {
  a:1,
  b:2,
  c:3
}
type objType = keyof typeof obj
```

but I think this might be just an array of allowed values instead of an object, so you could do the typescript sneaky move of `typeof {someArray}[number]`, like this:

```typescript
const arr = ['a', 'b', 'c'] as const
type arrType = typeof arr[number] // <= I think this is the bit that I might have shown once to @craigpalermo 
```